src/Unread.ts

```diff
@@ -59,36 +60,40 @@ export function doesRoomHaveUnreadMessages(room: Room): boolean {
         return false;
     }
 
+    for (const timeline of [room, ...room.getThreads()]) {
+        // If the current timeline has unread messages, we're done.
+        if (doesRoomOrThreadHaveUnreadMessages(timeline)) {
+            return true;
+        }
+    }
+    // If we got here then no timelines were found with unread messages.
+    return false;
+}
+
+export function doesRoomOrThreadHaveUnreadMessages(room: Room | Thread): boolean {
```

Comment:
The naming of this parameter is slightly confusing. I've seen you called it `roomOrThread` in other places. Could we rename it here too?